libraries/apollo-debug-server/src/commonMain/kotlin/com/apollographql/apollo3/debugserver/ApolloDebugServer.kt

```diff
@@ -0,0 +1,34 @@
+package com.apollographql.apollo3.debugserver
+
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.debugserver.internal.server.Server
+import com.apollographql.apollo3.debugserver.internal.server.createServer
+
+object ApolloDebugServer {
+  private val apolloClients = mutableMapOf<ApolloClient, String>()
+  private var server: Server? = null
+
+  fun registerApolloClient(apolloClient: ApolloClient, id: String = "client") {
+    if (apolloClients.containsKey(apolloClient)) error("Client '$apolloClient' already registered")
+    if (apolloClients.containsValue(id)) error("Name '$id' already registered")
+    apolloClients[apolloClient] = id
+    startOrStopServer()
+  }
+
+  fun unregisterApolloClient(apolloClient: ApolloClient) {
+    apolloClients.remove(apolloClient)
+    startOrStopServer()
+  }
+
+  private fun startOrStopServer() {
+    if (apolloClients.isEmpty()) {
+      server?.stop()
+    } else {
+      if (server == null) {
+        server = createServer(apolloClients).apply {
+          start()
+        }
+      }
+    }
+  }
```

Comment:
This doesn't work if you have 2 ApolloClients:

```kotlin
ApolloDebugServer.registerApolloClient(client1, "client1")
// server is started
ApolloDebugServer.registerApolloClient(client2, "client2")
// nothing happens
```

Reply:
Woops sorry I missed that comment 🙄 

But I think it works as expected? The server is already running, so no need to start it again. But it has a reference to the list of clients, so it will see it.

Reply:
Oooh I missed that the list was mutable. That's not super thread safe. Probably not going to be much a problem but if it ever becomes one, it'll be hell to investigate. Make the list an `AtomicReference<>` ?